packages/npm/@amazeelabs/react-framework-bridge/src/utils/__tests__/atomic.test.tsx

```diff
@@ -473,6 +474,121 @@ describe('Route rendering', () => {
       </div>
     `);
   });
+
+  it('allows to pass zustand stores into organisms', () => {
+    const store = createStore(() => ({
+      x: 'Foo',
+      y: 0,
+    }));
+    const { container } = render(
+      <Route
+        definition={Content}
+        intl={{ defaultLocale: 'en', locale: 'en' }}
+        input={{
+          intro: { title: 'Dynamic' },
+          body: [
+            {
+              key: 'dynamic',
+              input: {
+                dynamicData: store,
+              },
+            },
+          ],
+        }}
+      />,
+    );
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <div>
+          <div>
+            <h1>
+              Dynamic
+            </h1>
+          </div>
+          <div>
+            <p>
+              Foo
+            </p>
+          </div>
+        </div>
+      </div>
+    `);
+  });
+
+  it('updates organisms based on store changes', async () => {
+    vi.useFakeTimers();
+
+    type Store = {
+      x: string;
+      y: number;
+      swap: () => void;
+    };
+
+    const store = createStore<Store>((set) => ({
+      x: 'Foo',
+      y: 0,
+      swap: () => set({ x: 'Bar' }),
+    }));
+
+    setTimeout(() => {
+      act(() => {
+        store.getState().swap();
+      });
+    }, 50);
+
+    const { container } = render(
+      <Route
+        definition={Content}
+        intl={{ defaultLocale: 'en', locale: 'en' }}
+        input={{
+          intro: { title: 'Dynamic' },
+          body: [
+            {
+              key: 'dynamic',
+              input: {
+                dynamicData: store,
+              },
+            },
+          ],
+        }}
+      />,
+    );
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <div>
+          <div>
+            <h1>
+              Dynamic
+            </h1>
+          </div>
+          <div>
+            <p>
+              Foo
+            </p>
+          </div>
+        </div>
+      </div>
+    `);
+    vi.runAllTimers();
```

Comment:
Why it was required to use `setTimeout`? If we called `act(...)` right here, wouldn't it lead to the same result?